### ISSUE TITLE
Using transfomer causes NullReferenceException

### ISSUE THREAD

**Issue by OV2012** [Particular/NServiceBus.RavenDB#391]
hi, 
its look like that transofrmer causes nullreferenceexception with nservicebus.ravendb (4.2.4).
see exception callstack below.
and i've published an isolated example: https://github.com/OV2012/NServiceBus.RavenDB.Sample
or
https://github.com/OV2012/NServiceBus.RavenDB.Sample/blob/master/Server/PrintInvoiceHandler.cs

am i doing something wrong?
regards
viktor

```c#
public class PrintInvoiceHandler:IHandleMessages<PrintInvoice>
    {
        public async Task Handle(PrintInvoice message, IMessageHandlerContext context)
        {
            var session = context.SynchronizedStorageSession.RavenSession();

            //Upps!!!!
            var item = await session.LoadAsync<PrintInvoiceDataTransfomer.Result>(message.OrderId, typeof(PrintInvoiceDataTransfomer));

        }
}

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at NServiceBus.Persistence.RavenDB.TimeoutDataV1toV2Converter.IsTimeoutData(RavenJObject ravenJObject)
   at NServiceBus.Persistence.RavenDB.TimeoutDataV1toV2Converter.BeforeConversionToEntity(String key, RavenJObject document, RavenJObject metadata)
   at Raven.Client.Document.SessionOperations.LoadTransformerOperation.<ParseResults>d__5`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Raven.Client.Document.SessionOperations.LoadTransformerOperation.Complete[T](MultiLoadResult multiLoadResult)
   at Raven.Client.Document.Async.AsyncDocumentSession.<LoadUsingTransformerInternalAsync>d__69`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Raven.Client.Document.Async.AsyncDocumentSession.<LoadAsync>d__67`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Server.PrintInvoiceHandler.<Handle>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.SagaPersistenceBehavior.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.LoadHandlersConnector.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.InvokeSagaNotFoundBehavior.<Invoke>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.DeserializeLogicalMessagesConnector.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.Persistence.RavenDB.OpenAsyncSessionBehavior.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.SubscriptionReceiverBehavior.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.ReceivePerformanceDiagnosticsBehavior.<Invoke>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.ProcessingStatisticsBehavior.<Invoke>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.TransportReceiveToPhysicalMessageProcessingConnector.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.MainPipelineExecutor.<Invoke>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.ReceiveStrategy.<TryProcessMessage>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NServiceBus.SendsAtomicWithReceiveNativeTransactionStrategy.<ProcessMessage>d__2.MoveNext()

**Comment by mauroservienti** [comment]
Hi @OV2012,

many thanks for the repro. I believe I know what's going on. The [TimeoutDataV1toV2Converter](https://github.com/Particular/NServiceBus.RavenDB/blob/develop/src/NServiceBus.RavenDB/Timeouts/TimeoutDataV1toV2Converter.cs#L31) starts from the assumption that what's coming from the server is a document, in which case it has `metadata`. When there is a `transformer` in the middle `metadata` can be null when the converter is invoked.

I'll validate my assumptions and if that's the case I'll start working on a fix.

**Comment by OV2012** [comment]
your assumption is correct!
here is the fix:)

` var clrMetaData = ravenJObject?["Raven-Clr-Type"];`

and quick&dirty test:

```
namespace NServiceBus.RavenDB.Tests.Timeouts
{
    using System;
    using System.Linq;
    using System.Threading.Tasks;
    using NServiceBus.Persistence.RavenDB;
    using NUnit.Framework;
    using Raven.Client.Indexes;

    [TestFixture]
    public class When_using_transformers_with_timeoutdatav1tov2converter : RavenDBPersistenceTestBase
    {
        public override void SetUp()
        {
            base.SetUp();

            new DummyDataTransfomer().Execute(store);

            store.Listeners.RegisterListener(new FakeLegacyTimoutDataClrTypeConversionListener());
            store.Listeners.RegisterListener(new TimeoutDataV1toV2Converter());
        }

        [Test]
        public async Task Should_not_fail()
        {
           var session = store.OpenAsyncSession();

            await session.StoreAsync(new Dummy(){Id = "dummies/test"});
            await session.SaveChangesAsync();

            var result = await session.LoadAsync<DummyDataTransfomer.Result>("dummies/test", typeof(DummyDataTransfomer));
            Assert.IsNotNull(result);
        }

        public class Dummy
        {
            public string Id { get; set; }
        }


        public class DummyDataTransfomer : AbstractTransformerCreationTask<Dummy>
        {
            public class Result
            {
                public Guid Id { get; set; }
                public DateTime DoneAt { get; set; }
            }

            public DummyDataTransfomer()
            {
                TransformResults = results => from item in results
                    select new
                    {
                        OrderId = item.Id
                    };
            }
        }
    }
}
```

**Comment by mauroservienti** [comment]
Many thanks for the failing test, I’ll take care of implementing the fix.

**Comment by mauroservienti** [comment]
@OV2012 I raised https://github.com/Particular/NServiceBus.RavenDB/pull/392 to fix this. Thanks!

**Comment by mauroservienti** [comment]
Fixed by https://github.com/Particular/NServiceBus.RavenDB/pull/392

**Comment by mauroservienti** [comment]
@OV2012 this is now fixed and released as 4.2.5: https://github.com/Particular/NServiceBus.RavenDB/releases/tag/4.2.5

**Comment by OV2012** [comment]
@mauroservienti  thank you for your quick reaction!!
